plugins/opensslcryptoplugin/evp/evp_helpers_p.h

```diff
@@ -60,9 +60,8 @@ struct CipherSessionDataDeleter
         if (csd->evp_cipher_ctx) {
             EVP_CIPHER_CTX_free(csd->evp_cipher_ctx);
         }
-        if (csd->evp_md_ctx) {
-            EVP_MD_CTX_destroy(csd->evp_md_ctx);
-        }
+        // NOTE: csd->evp_md_ctx is already destroyed,
+        //       the _finalize functions will destroy it normally, or the _update ones too if there are errors.
```

Comment:
What about the case where the session is destroyed due to timeout, rather than a call to finalize?

Reply:
You are right, I ought to fix this.